### PR TITLE
Map ur_bool_t to cl_bool in sampler getinfo for opencl adapter

### PR DESCRIPTION
**Pull request by omarahmed1111**
[INTEL/llvm](https://github.com/intel/llvm/pull/14769)

### CODE REVIEW

File: source/adapters/opencl/sampler.cpp [hunk 1]
```diff
@@ -159,6 +159,12 @@ urSamplerGetInfo(ur_sampler_handle_t hSampler, ur_sampler_info_t propName,
                 sizeof(ur_sampler_addressing_mode_t));
 
   size_t CheckPropSize = 0;
+  // ur_bool_t have a size of uint8_t, but cl_bool size have the size of
+  // uint32_t so this makes a mapping between them.
+  if (propName == UR_SAMPLER_INFO_NORMALIZED_COORDS &&
+      propSize == sizeof(ur_bool_t)) {
+    propSize = sizeof(cl_bool);
```
Comment (aarongreig) [inline source/adapters/opencl/sampler.cpp:166]: I think we want an entirely different path for `INFO_NORMALIZED_COORDS`, this is passing `pPropValue` directly into `clGetSamplerInfo` and returning the property size we get from CL directly in `pPropSizeRet`. We should really be passing a `cl_bool*` to the CL api and converting it to `ur_bool_t` to be returned, as well as returning `sizeof(ur_bool_t)` for `pPropSizeRet`

Comment (omarahmed1111) [reply source/adapters/opencl/sampler.cpp]: Good point, I changed that. however, not sure if I handled it in the most efficient way (redundant code I mean) so let me know if I could optimize this code more.

File: source/adapters/opencl/sampler.cpp [hunk 2]
```diff
@@ -158,16 +158,37 @@ urSamplerGetInfo(ur_sampler_handle_t hSampler, ur_sampler_info_t propName,
...
+    cl_bool normalized_coords = false;
+    Err = mapCLErrorToUR(
+        clGetSamplerInfo(cl_adapter::cast<cl_sampler>(hSampler), SamplerInfo,
+                         sizeof(cl_bool), &normalized_coords, nullptr));
+    if (pPropValue && propSize != sizeof(ur_bool_t)) {
+      return UR_RESULT_ERROR_INVALID_SIZE;
+    }
+    UR_RETURN_ON_FAILURE(Err);
+    if (pPropValue) {
+      *static_cast<ur_bool_t *>(pPropValue) = (ur_bool_t)normalized_coords;
```
Comment (aarongreig) [inline source/adapters/opencl/sampler.cpp:175]:
```suggestion
      *static_cast<ur_bool_t *>(pPropValue) = static_cast<ur_bool_t>(normalized_coords);
```

File: source/adapters/opencl/sampler.cpp (hunk 1 above)
Comment (aarongreig) [reply source/adapters/opencl/sampler.cpp]: nah looks good, we have similar code to deal with edge cases elsewhere in the CL adapter

File: source/adapters/opencl/sampler.cpp (hunk 2 above)
Comment (omarahmed1111) [reply source/adapters/opencl/sampler.cpp]: Fixed

**Review by aarongreig — APPROVED** [review]

**Pull request merged**